lib/ansible/playbook/role/definition.py

```diff
@@ -139,6 +142,30 @@ def _load_role_path(self, role_name):
         append it to the default role path
         '''
 
+        # create a templar class to template the dependency names, in
+        # case they contain variables
+        if self._variable_manager is not None:
+            all_vars = self._variable_manager.get_vars(play=self._play)
+        else:
+            all_vars = dict()
+
+        templar = Templar(loader=self._loader, variables=all_vars)
+        role_name = templar.template(role_name)
+
+
+        # FIXME: sanity check the name first; don't try this with path-ish role names, etc
+        # try to load as a collection-based role first
+        role_tuple = get_collection_role_path(role_name, self._collection_list)
+
+        if role_tuple:
+            # we found it, just return what we got
+            return role_tuple
+
+        # FIXME: refactor this to be callable from internal so we can properly order ansible.legacy searches with the
+        # collections keyword
+        if self._collection_list and not 'ansible.legacy' in self._collection_list:
```

Comment:
It's always a list, no additional check is needed + it's preferable to use `not in`
```suggestion
        if 'ansible.legacy' not in self._collection_list:
```

Reply:
used `not in`, but there will likely be valid situations where it's `None`, so still need the first part of the conditional